### PR TITLE
Let `suspenders --api` work

### PR DESCRIPTION
**Pull request by gabebw**
When running `suspenders --api`, there are some missing files in the
generated API-only Rails app compared to the non-API version, which
causes the suspenders generator to crash.

To fix this, just don't do anything when those files aren't found.

* There are no HTTP error pages, so `customize_error_pages` fails
* `config/initializers/assets.rb` is removed by the API generator, leading to this error:

```
    Setting up the production environment
          create  config/smtp.rb
        prepend  config/environments/production.rb
          insert  config/environments/production.rb
          append  config/environments/production.rb
          insert  config/environments/production.rb
          insert  config/environments/production.rb
    Traceback (most recent call last):
            25: from /Users/gabe/.rbenv/versions/2.5.0/bin/suspenders:23:in `<main>'
            24: from /Users/gabe/.rbenv/versions/2.5.0/bin/suspenders:23:in `load'
            23: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/bin/suspenders:23:in `<top (required)>'
            22: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
            21: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/group.rb:232:in `dispatch'
            20: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:133:in `invoke_all'
            19: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:133:in `map'
            18: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:133:in `each'
            17: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:133:in `block in invoke_all'
            16: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
            15: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
            14: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/lib/suspenders/generators/app_generator.rb:39:in `finish_template'
            13: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:115:in `invoke'
            12: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/group.rb:230:in `dispatch'
            11: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
            10: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
            9: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/lib/suspenders/generators/app_generator.rb:46:in `suspenders_customization'
            8: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:115:in `invoke'
            7: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/group.rb:230:in `dispatch'
            6: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
            5: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
            4: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/lib/suspenders/generators/app_generator.rb:99:in `setup_production_environment'
            3: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/railties-5.1.5/lib/rails/generators/app_base.rb:151:in `build'
            2: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/lib/suspenders/app_builder.rb:135:in `setup_asset_host'
            1: from /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/lib/suspenders/actions.rb:5:in `replace_in_file'
    /Users/gabe/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/suspenders-1.46.0/lib/suspenders/actions.rb:5:in `read': No such file or directory @ rb_sysopen - /Users/gabe/code/personal/friendlyvents/config/initializers/assets.rb (Errno::ENOENT)
```

### CODE REVIEW

**Comment by gabebw** [comment]
I realize that Suspenders is not an API-first generator, but I like its defaults and I think it's better to let it  run, then edit it, rather than have it crash when passed the `--api` flag.

File: lib/suspenders/app_builder.rb [hunk 1]
```diff
@@ -120,9 +120,11 @@ def setup_asset_host
         "# config.action_controller.asset_host = 'http://assets.example.com'",
         'config.action_controller.asset_host = ENV.fetch("ASSET_HOST", ENV.fetch("APPLICATION_HOST"))'
 
-      replace_in_file 'config/initializers/assets.rb',
-        "config.assets.version = '1.0'",
-        'config.assets.version = (ENV["ASSETS_VERSION"] || "1.0")'
+      if File.exist?("config/initializers/assets.rb")
+        replace_in_file 'config/initializers/assets.rb',
```
Comment (gabebw) [inline lib/suspenders/app_builder.rb]: I think an even better solution to this problem would be adding the `if File.exist?(path)` check in `replace_in_file`, which would also fix the errors that occur when running `suspenders --skip-action-mailer`.

**Comment by gabebw** [comment]
@iwz Thanks for merging! 🎉

**Comment by mike-burns** [comment]
@gabebw and take note that there are a growing number of things you can remove after the fact using `rails destroy suspenders:...`!

**Comment by BenMusch** [comment]
Sorry if this is the wrong place to ask -- Is there a plan to include this in a new version?

**Comment by mike-burns** [comment]
Indeed, it has been a while since we've cut a release. Yes, we would like to do that soon.

**Comment by mike-burns** [comment]
@BenMusch 1.47.0 is now released. Thank you for the prodding!

**Pull request merged**